lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -1363,15 +1543,7 @@ protected function get_layout_styles( $block_metadata ) {
 							$selector,
 							$class_name
 						);
-						$block_rules    .= static::to_ruleset(
-							$layout_selector,
-							array(
-								array(
-									'name'  => 'display',
```

Comment:
While I think of it, one potential issue with this one is that `display` is being manually injected here because it is not included in the allow-listed list of CSS properties (here's the core PR: https://github.com/WordPress/wordpress-develop/pull/2928 where `display` was discussed), so it has special handling here to determine that the value is from a hard-coded list of valid display modes.

Do we need to have a mechanism to allow potentially unsafe declarations to bypass calling `safecss_filter_attr`? I'm imagining some kind of approach where we can say to the style engine (at some point), _here is an already validated declaration, please trust it_ 😅

Reply:
> Do we need to have a mechanism to allow potentially unsafe declarations to bypass calling `safecss_filter_attr`?

We kind of already have that... 😉 I was allowing CSS variables in the `Declarations` object in this PR, so I added `display` when the value is not `none` (I believe `display:none` is the reason we don't allow `display` in safe-css, so this implementation accounts for that scenario)

Reply:
For context, check out the tweaks in `packages/style-engine/class-wp-style-engine-css-declarations.php` in this PR. If the property starts with `--` it skips `safecss_filter_attr()`, (to account for CSS variables), and the same thing happens for `display` if the value is not `none` 👍